.pre-commit-config.yaml

```diff
@@ -1,123 +1,12 @@
-        #                 --show-error-codes,
-        #                 --no-warn-unused-ignores,  # Ignore 'type: ignore' comments that are not used.
-        #                 --disable-error-code=attr-defined,  # "Module has no attribute 'XXX'" occurs because of the pre-commit env.
-        #                 --disable-error-code=name-defined,  # "Name 'XXX' is not defined" occurs because of the pre-commit env.
-        #                 --disable-error-code=var-annotated,
-        #                 --disable-error-code=no-any-return
-
-        #           ]
-        #           additional_dependencies: [tokenize-rt==3.2.0, numpy==1.22]
-        #           files: ^(usgsxplore|tests|doc/code)
-
-
-        # Sort imports using isort
-        - repo: https://github.com/PyCQA/isort
-          rev: 5.12.0
-          hooks:
-                  - id: isort
-                    args: ["--profile", "black"]
-
-        # Automatically upgrade syntax to a minimum version
-        - repo: https://github.com/asottile/pyupgrade
-          rev: v3.15.0
-          hooks:
-                - id: pyupgrade
-                  args: [--py310-plus]
-
-        # Various formattings
-        - repo: https://github.com/pre-commit/pygrep-hooks
-          rev: v1.10.0
-          hooks:
-                # Single backticks should apparently not be used
-                - id: rst-backticks
-                # Check that all directives end with double colon
-                - id: rst-directive-colons
-                  types: [text]
-                  types_or: [python, rst]
-                # Inline code should not touch normal text
-                - id: rst-inline-touching-normal
-                  types: [text]
-                  types_or: [python, rst]
-                # # Eval should never be used (can do arbitrary code execution)
-                # - id: python-no-eval
-                # Enforce the use of type annotations instead of docstring type comments
-                - id: python-use-type-annotations
-
-        # Add custom regex lints (see .relint.yml)
-        # - repo: https://github.com/codingjoe/relint
-        #   rev: 2.0.0
-        #   hooks:
-        #         - id: relint
-        # - repo: local
-        #   hooks:
-        #         # Generate pip's requirements.txt from conda's environment.yml to ensure consistency
-        #         - id: pip-to-conda
-        #           name: Generate pip dependency from conda
-        #           language: python
-        #           entry: .github/scripts/generate_pip_deps_from_conda.py
-        #           files: ^(environment.yml|requirements.txt)$
-        #           pass_filenames: false
-        #           additional_dependencies: [tomli, pyyaml]
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.11.7
+  hooks:
+    # Run the linter.
+    - id: ruff
```

Comment:
do you know what is the max line length allowed by Ruff by default? For black, we had a limit of 120 characters, not the default 79.